### ISSUE TITLE
FR:failed connecting to the  tailscale service :windows 11

### ISSUE THREAD

**Issue by leonaholi** [tailscale/tailscale#13494]
### What are you trying to do?

when tailscale is launched above error is shown :
tried: update of tailscale to version 1.74.0(t280bab29-gbfc27063c)
still, same error reboot was done to Windows about 10 times.
tried to update all the packages in Windows with {winget upgrade --include-unknown --all}
tried to downgrade the tailscale client to an older version no change note still same same error.

       path to tailscaled socket
PS C:\Windows\system32> tailscale bugreport
503 Service Unavailable: no backend
PS C:\Windows\system32>

the service is running :
![image](https://github.com/user-attachments/assets/8ed39647-6d94-4c2d-bb45-84578e220ff8)

### How should we solve this?

I appreciate any help..

### What is the impact of not solving this?

unable to launch  tailscale.

**Comment by vuanhson** [comment]
It's a bug report, not FR/Feature Request 😄

**Comment by leonaholi** [comment]
![image](https://github.com/user-attachments/assets/dc2a2e0c-dcd0-440f-b6a0-1384f920efd3)
yes bug report

**Comment by leonaholi** [comment]
PS C:\Windows\system32> tailscale status
failed to connect to local tailscaled (which appears to be running as tailscaled.exe, pid 11700). Got error: 503 Service Unavailable: no backend
PS C:\Windows\system32> tailscale.exe status
PS C:\Windows\system32> tailscaled.exe status
2024/09/19 16:54:55 tailscaled does not take non-flag arguments: ["status"]
PS C:\Windows\system32> tailscaled.exe
logtail started
local disk logdir: C:\ProgramData\Tailscale\Logs
Program starting: v1.74.0-t28e0bab29-gbfc27063c, Go 1.23.1: []string{"C:\\Program Files\\Tailscale\\tailscaled.exe"}
failed to create log file: open C:\ProgramData\Tailscale\Logs\tailscale-service-20240919T165505-1726754105.txt: Access is denied.
LogID: 9f9b443b2109abe5756f00b34dda27922534e456bf3c85f81e06615e39d641c6
failed to create log file: open C:\ProgramData\Tailscale\Logs\tailscale-service-20240919T165505-1726754105.txt: Access is denied.
logpolicy: using dir C:\ProgramData\Tailscale
logpolicy.ConfigFromFile C:\ProgramData\Tailscale\tailscaled.log.conf: open C:\ProgramData\Tailscale\tailscaled.log.conf: The system cannot find the file specified.
logpolicy.Config.Validate for C:\ProgramData\Tailscale\tailscaled.log.conf: config is nil
logpolicy.Config.Save for C:\ProgramData\Tailscale\tailscaled.log.conf: open C:\ProgramData\Tailscale\tailscaled.log.conf.tmp2206166807: Access is denied.
failed to create log file: open C:\ProgramData\Tailscale\Logs\tailscale-service-20240919T165505-1726754105.txt: Access is denied.
dns: using *dns.windowsManager
failed to create log file: open C:\ProgramData\Tailscale\Logs\tailscale-service-20240919T165505-1726754105.txt: Access is denied.
dns down: open C:\Windows\system32\drivers\etc\hosts.tmp59069821: Access is denied.
failed to create log file: open C:\ProgramData\Tailscale\Logs\tailscale-service-20240919T165505-1726754105.txt: Access is denied.
WSL: could not list distributions: wslRun([wsl.exe -l]): exit status 1: ""
failed to create log file: open C:\ProgramData\Tailscale\Logs\tailscale-service-20240919T165505-1726754105.txt: Access is denied.
flushing log.
failed to create log file: open C:\ProgramData\Tailscale\Logs\tailscale-service-20240919T165505-1726754105.txt: Access is denied.
logger closing down
safesocket.Listen: namedpipe.Listen: open \\.\pipe\ProtectedPrefix\Administrators\Tailscale\tailscaled: Access is denied.
failed to create log file: open C:\ProgramData\Tailscale\Logs\tailscale-service-20240919T165505-1726754105.txt: Access is denied.
PS C:\Windows\system32>

**Comment by pirafrank** [comment]
Adding to this. Upgraded from 1.64 to 1.74 and in the same situation. Checking status outputs this;

```
PS C:\Users\francesco ~ tailscale status
failed to connect to local tailscaled (which appears to be running as tailscaled.exe, pid 40804). Got error: Get "http://local-tailscaled.sock/localapi/v0/status": EOF
```

**Comment by caiyesd** [comment]
I meet this issue as well.

**Comment by pirafrank** [comment]
Hi, this is still of an issue after >1 month and several updates to the client. None of them fixed the issue. Restarting doesn't help and it's now somehow permanently broken.

**Comment by leonaholi** [comment]
Still broken will have to fully reset windows if the issue is persistent. Plan is to reset tonight +3GMT

**Comment by stvnksslr** [comment]
been experiencing this issue for sometime as well on the latest tailscale and have tried uninstall/reinstalling many times nothing seems to be able to get it to work since some windows update around September.

tailscale: 1.76.6
windows 11 
23H2

**Comment by leonaholi** [comment]
Fix is .
1. full reset of windows 11
2. optionally uninstall of wireguard driver 
3. Reinstall of tailscale and adopt to your network.

Hope it helps...

**Comment by McHauge** [comment]
Commenting to say I have this issues atm aswell, will try removing wireguard and re installing tailscale. not that I was using wireguard

**Comment by McHauge** [comment]
SOLVED on my end, got it working without re-installing.

Here is the steps I did:

1. remove wireguard
2. uninstall tailscale
3. clear programdata/tailscale
4. clear appdata/local/tailscale
5. reboot and reinstall tailscale

run tailscale -> issues still percists

1. close all instances of tailscale including the tailscale service
2. open powershell in admin
3. run tailscaled.exe

look for no access errors
if you have acess errors it might not have closed fully yet or you might need different premissions for the service.
(I setup mine to run as the local admin, most likely not needed)

if it runs with no "access denied" errors and you get something like this:

```
PS C:\Windows\system32> tailscaled.exe
logtail started
local disk logdir: C:\ProgramData\Tailscale\Logs
Program starting: v1.76.6-t1edcf9d46-gd0a6cd8b2, Go 1.23.1: []string{"C:\\Program Files\\Tailscale\\tailscaled.exe"}
LogID: ---
logpolicy: using dir C:\ProgramData\Tailscale
dns: using *dns.windowsManager
running ipconfig /registerdns ...
wgengine.NewUserspaceEngine(tun "Tailscale") ...
Installing driver 0.14
Extracting driver
WSL: found 2 distributions
Installing driver
Creating adapter
ran ipconfig /registerdns in 3.032s
running ipconfig /flushdns ...
ran ipconfig /flushdns in 16ms
Timed out waiting for device query: The wait operation timed out. (Code 0x00000102)
Failed to setup adapter (problem code: 0x1C, ntstatus: 0xC0000490): There was no match for the specified key in the index. (Code 0x00000491)
Installing driver 0.14
Extracting driver
Installing driver
Creating adapter
Timed out waiting for device query: The wait operation timed out. (Code 0x00000102)
Failed to setup adapter (problem code: 0x1C, ntstatus: 0xC0000490): There was no match for the specified key in the index. (Code 0x00000491)
```

then you can continue following the guide from here:
(https://github.com/tailscale/tailscale/issues/6461)

run these in an admin terminal or admin powershell

1. Export the list of drivers in the table form to a text file using the command:
dism /online /get-drivers /format:table > c:\drivers.txt

2. Find wintun or wireguard from the list, and note the "Published Name", oemNN.inf

3. Remove the driver using the command:
pnputil.exe /d oemNN.inf

(for me the actual wireguard driver was called: oem119.inf)

after this run tailscaled.exe again and it should now be able to install the driver 

if it install correctly close the powershell, remember to stop it first: ctrl+c
then start the service via the service manager

After this run the tailscale app like normal and login again, and it should now work as normal

**Comment by McHauge** [comment]
Welp that was a short sycsess, it's now back to being broken again after having turned tailscale on and off a couple of times via this button:
![image](https://github.com/user-attachments/assets/cc3dbff8-e4fd-4451-99e5-1aee587376a3)

So idk I'm guessing the service gets stuck and never fully closes before starting up again thus locking it out. or something, idk. I'll try my own fix again. but not sure if that's gonna be a good long term solution.

I have the log files from both cases, if that would help a dev, just not sure if they are "safe" to post here in raw format

**Comment by McHauge** [comment]
hmm this time it seems I only need to stop backend service and restart it manually via an admin powershell

**Comment by pirafrank** [comment]
I resolved by uninstall Tailscale completely, deleting any related folder in AppData and relative subdirs, rebooting, installing it again, rebooting once again.

**Comment by dblohm7** [comment]
@leonaholi was this issue specifically when upgrading via WinGet?

**Comment by zilpe** [comment]
@dblohm7 I'm experiencing the exact same issue with the Windows installer.

**Comment by McHauge** [comment]
> @leonaholi was your issue specifically when upgrading via WinGet? I'm trying to figure out whether we need to focus this issue on WinGet specifically and move other reports to a separate issue if unrelated.

my issues were with the Win.exe installer, but I have not had the issue come up after the deletion of the wireguard driver, shown in my list of steps :)

**Comment by 15169670031** [comment]
终于解决了，因为卸载vmware后并没有清除物理网卡上的VMware Bridge Protocol 驱动，并且无法直接删除，我是参考这个博客清除后解决：https://blog.csdn.net/weixin_42550185/article/details/144368669

**Comment by wlin16** [comment]
Fixed by synchronising the system time :(((((((((((

**Comment by adityakiwi** [comment]
im having the same issue now on windows arm

**Comment by nickkhyl** [comment]
@adityakiwi: Can you please [generate a bug report](https://tailscale.com/kb/1227/bug-report?tab=windows) and post it here?

**Comment by bryanc0033** [comment]
Same issue also on Windows 10 22H2 32-bit with last TS update.  Tried uninstall/reinstall which didn't work.

**Comment by neqq3** [comment]
> 终于解决了，因为卸载vmware后并没有清除物理网卡上的VMware Bridge Protocol 驱动，并且无法直接删除，我是参考这个博客清除后解决：https://blog.csdn.net/weixin_42550185/article/details/144368669

真的有用，感谢，我直接从第六步开始——6)、到这里[官方]Wise Registry Cleaner – 免费的注册表清理工具
清理完还没重启就已经连上了。

**Comment by DevDorrejo** [comment]
Have this issue, with default installation, last windows 11 with all updates installed. #17104

**Comment by keirannnnnnnn** [comment]
Same issue starting to happen on our Windows Server 2025 servers after they have been restarted, we have been rolling software back to 1.78 and disabling tailscale updates for now, its not ideal but its a temp solution at least.

**Comment by starlit-rocketship** [comment]
FIX FOUND!

I used REVO Uninstaller to remove it. Select advanced and select all traces and delete them.

Then when it is uninstalled, you can reinstall as normal and it gave me the log in screen

**Comment by Ghosty-Gigabytes** [comment]
> I used REVO Uninstaller to remove it. Select advanced and select all traces and delete them.
> 
> Then when it is uninstalled, you can reinstall as normal and it gave me the log in screen

Did you use the free version or pro version, 'cause I just tried the free version but still Tailscale didn't get fully purged (no prompt for login) and the fault wasn't resolved.

**Comment by starlit-rocketship** [comment]
> > I used REVO Uninstaller to remove it. Select advanced and select all traces and delete them.
> > Then when it is uninstalled, you can reinstall as normal and it gave me the log in screen
> 
> Did you use the free version or pro version, 'cause I just tried the free version but still Tailscale didn't get fully purged (no prompt for login) and the fault wasn't resolved.

Just the free version. 

When you click uninstall of tailscale, there will be 2 screens that pop up about additional files it finds. 

I select Advanced scan at start. Then select all & delete when it lists the registry files etc it finds

Hope it helps you

**Comment by DevDorrejo** [comment]
did not work in my side.

**Comment by geoffreyIOOF** [comment]
Have you seen this issue before. 
I uninstall using revo uninstaller, and also tried to remove the wintun driver (it get reinstall after starting tailscaled as expected), but no luck I still get a "failed connecting to the tailscale service".

Any Advice ?

wgengine.NewUserspaceEngine(tun "Tailscale") ...
2025-10-27T14:05:38.086+10:00: Using existing driver 0.14
2025-10-27T14:05:38.117+10:00: Creating adapter
2025-10-27T14:05:53.148+10:00: Timed out waiting for device query: The wait operation timed out. (Code 0x00000102)
2025-10-27T14:05:53.148+10:00: Failed to setup adapter (problem code: 0x38, ntstatus: 0x0): The device is not ready for use. (Code 0x000010DF)
2025-10-27T14:05:53.282+10:00: State of service "NetSetupSvc" and its dependencies:
2025-10-27T14:05:53.283+10:00: [
2025-10-27T14:05:53.283+10:00:   {
2025-10-27T14:05:53.283+10:00:     "serviceName": "NetSetupSvc",
2025-10-27T14:05:53.283+10:00:     "serviceType": "WIN32",
2025-10-27T14:05:53.283+10:00:     "state": "RUNNING",
2025-10-27T14:05:53.283+10:00:     "startupType": "DEMAND_START",
2025-10-27T14:05:53.283+10:00:     "triggers": [
2025-10-27T14:05:53.283+10:00:       {
2025-10-27T14:05:53.283+10:00:         "triggerType": 7,
2025-10-27T14:05:53.283+10:00:         "action": 1,
2025-10-27T14:05:53.283+10:00:         "triggerSubtype": "{2D7A2816-0C5E-45FC-9CE7-570E5ECDE9C9}",
2025-10-27T14:05:53.283+10:00:         "dataItems": [
2025-10-27T14:05:53.283+10:00:           {
2025-10-27T14:05:53.283+10:00:             "dataType": 1,
2025-10-27T14:05:53.283+10:00:             "data": "7508bca3230a8f12"
2025-10-27T14:05:53.283+10:00:           }
2025-10-27T14:05:53.283+10:00:         ]
2025-10-27T14:05:53.283+10:00:       },
2025-10-27T14:05:53.283+10:00:       {
2025-10-27T14:05:53.283+10:00:         "triggerType": 7,
2025-10-27T14:05:53.283+10:00:         "action": 1,
2025-10-27T14:05:53.283+10:00:         "triggerSubtype": "{2D7A2816-0C5E-45FC-9CE7-570E5ECDE9C9}",
2025-10-27T14:05:53.283+10:00:         "dataItems": [
2025-10-27T14:05:53.283+10:00:           {
2025-10-27T14:05:53.283+10:00:             "dataType": 1,
2025-10-27T14:05:53.283+10:00:             "data": "7510bca3230a8f12"
2025-10-27T14:05:53.283+10:00:           }
2025-10-27T14:05:53.284+10:00:         ]
2025-10-27T14:05:53.284+10:00:       },
2025-10-27T14:05:53.284+10:00:       {
2025-10-27T14:05:53.284+10:00:         "triggerType": 6,
2025-10-27T14:05:53.284+10:00:         "action": 1,
2025-10-27T14:05:53.284+10:00:         "triggerSubtype": "{BC90D167-9470-4139-A9BA-BE0BBBF5B74D}",
2025-10-27T14:05:53.284+10:00:         "dataItems": [
2025-10-27T14:05:53.284+10:00:           {
2025-10-27T14:05:53.284+10:00:             "dataType": 2,
2025-10-27T14:05:53.284+10:00:             "data": "610031003100310066003100630035002d0035003900320033002d0034003700630030002d0039006100360038002d006400300062006100660062003500370037003900300031000000"
2025-10-27T14:05:53.284+10:00:           }
2025-10-27T14:05:53.284+10:00:         ]
2025-10-27T14:05:53.284+10:00:       }
2025-10-27T14:05:53.284+10:00:     ]
2025-10-27T14:05:53.284+10:00:   },
2025-10-27T14:05:53.284+10:00:   {
2025-10-27T14:05:53.284+10:00:     "serviceName": "RpcSs",
2025-10-27T14:05:53.284+10:00:     "serviceType": "WIN32",
2025-10-27T14:05:53.284+10:00:     "state": "RUNNING",
2025-10-27T14:05:53.284+10:00:     "startupType": "AUTO_START"
2025-10-27T14:05:53.284+10:00:   },
2025-10-27T14:05:53.284+10:00:   {
2025-10-27T14:05:53.284+10:00:     "serviceName": "DcomLaunch",
2025-10-27T14:05:53.284+10:00:     "serviceType": "WIN32",
2025-10-27T14:05:53.284+10:00:     "state": "RUNNING",
2025-10-27T14:05:53.284+10:00:     "startupType": "AUTO_START"
2025-10-27T14:05:53.284+10:00:   },
2025-10-27T14:05:53.284+10:00:   {
2025-10-27T14:05:53.284+10:00:     "serviceName": "RpcEptMapper",
2025-10-27T14:05:53.284+10:00:     "serviceType": "WIN32",
2025-10-27T14:05:53.284+10:00:     "state": "RUNNING",
2025-10-27T14:05:53.284+10:00:     "startupType": "AUTO_START"
2025-10-27T14:05:53.284+10:00:   }
2025-10-27T14:05:53.284+10:00: ]
2025-10-27T14:05:53.284+10:00: tstunNew: [v1] backoff: 6 msec

**Comment by geoffreyIOOF** [comment]
I finally fixed the issue following this video https://www.youtube.com/watch?v=yidWdy-Xwdk 
Reinstalling windows keeping files and apps.
Tailscale is back to work

**Comment by horstzipfel** [comment]
I don't think it's acceptable to reinstall windows. Does anyone has any other solution to this?

**Comment by geoffreyIOOF** [comment]
> I don't think it's acceptable to reinstall windows. Does anyone has any other solution to this?

Yeah It seems extreme, but this reinstall does not remove anything it keeps everything but reinstall windows files, and fix all the corrupted files.
And I dont think Tailscale is the cause of the issue, but a consequence of some corrupted files in windows.

**Comment by nickkhyl** [comment]
```
2025-10-27T14:05:53.148+10:00: Timed out waiting for device query: The wait operation timed out. (Code 0x00000102)
2025-10-27T14:05:53.148+10:00: Failed to setup adapter (problem code: 0x38, ntstatus: 0x0): The device is not ready for use. (Code 0x000010DF)
```

A common cause of this error code is a third-party component, such as an [NDIS Notify Object](https://learn.microsoft.com/en-us/windows-hardware/drivers/network/about-notify-objects), that was previously installed but improperly removed. For example, some users have encountered this issue after deleting VMware Workstation from `%programfiles%` instead of properly uninstalling it, or after installing it on a removable drive and then disconnecting it. In cases like this, certain files remain registered with the OS, and it fails installing or configuring network adapters due to a missing file.

The script below can help identify certain missing components:
```ps1
$keyPaths = "HKLM:\SYSTEM\CurrentControlSet\Control\Network\*\*"
$keys = Get-ChildItem -Path $keyPaths -ErrorAction SilentlyContinue
foreach ($key in $keys) {
     $ndiKeyPath = $key.PSPath + "\NDI"
     $clsid = (Get-ItemProperty -Path $ndiKeyPath -ErrorAction SilentlyContinue).CLSID
     if ($null -ne $clsid) {
         $componentId = (Get-ItemProperty -Path $key.PSPath -ErrorAction SilentlyContinue).ComponentId

         # Determine the path to the In-Proc COM Server .DLL
         $dllpath = ""
         try {
             $inprocServer32path = "HKLM:\SOFTWARE\Classes\CLSID\$clsid\InprocServer32"
             $dllpath = (Get-ItemProperty -Path $inprocServer32path)."(Default)"
         } catch {
             Write-Output "WARNING: The DLL registration not found for $clsid ($componentId)"
             continue
         }

         # Check if the path exists
         if (Test-Path -Path $dllpath) {
             Write-Output "Found the DLL for $clsid ($componentId) at $dllpath"
         } else {
             Write-Output "WARNING: The DLL for $clsid ($componentId) does not exist at $dllpath"
         }
     }
}
```

**Comment by horstzipfel** [comment]
Thank you very much for your replies. I had indeed some trouble with network adapters and resetting them via Windows11 function as described at https://www.howtogeek.com/how-to-reset-all-network-settings-in-windows-11/ fixed it for me.

**Comment by DevDorrejo** [comment]
Hello, pardon my ignorance, @nickkhyl .

I executed the script and show me this:

<img width="803" height="59" alt="Image" src="https://github.com/user-attachments/assets/6f9bc946-e34d-49ac-b944-e78395c57061" />

have I to remove those files?

**Comment by nickkhyl** [comment]
@DevDorrejo: You should not remove those files.

The script prints all NDIS notify object components installed on the device, both system and third-party.
These two are perfectly fine and aren't expected to cause any issues.

On some environments, the script may show warnings indicating that a component is still registered even though the corresponding .DLL file is missing. This usually happens due to incomplete or incorrect removal of third-party applications, such as VMware Workstation. Based on the script output, that's not the case here.

Additionally, some third-party components may contain bugs, but that's not the case on your environment either. The two listed files are well-known system components.

I'd suggest reaching out to support@tailscale.com

**Comment by horstzipfel** [comment]
Just to add on this, because it might also affect others: I recently removed TPM and information saved by tailscale could not be retrieved. My error message was:
> getLocalBackend error: store.New: failed to unseal state file: failed to unseal encryption key with TPM: tpm2.Load: TPM_RC_INTEGRITY (handle 0): integrity check failed

Removing the folder "C:\ProgramData\Tailscale" and restarting tailscale allowed me to reauthenticate and it seems stable now.

**Comment by HoustonBoston** [comment]
> Just to add on this, because it might also affect others: I recently removed TPM and information saved by tailscale could not be retrieved. My error message was:
> 
> > getLocalBackend error: store.New: failed to unseal state file: failed to unseal encryption key with TPM: tpm2.Load: TPM_RC_INTEGRITY (handle 0): integrity check failed
> 
> Removing the folder "C:\ProgramData\Tailscale" and restarting tailscale allowed me to reauthenticate and it seems stable now.

This worked!!! Thanks a lot!

**Comment by Rockin-Rog** [comment]
> FIX FOUND!
> 
> I used REVO Uninstaller to remove it. Select advanced and select all traces and delete them.
> 
> Then when it is uninstalled, you can reinstall as normal and it gave me the log in screen

THANKS...the Free Pro version of Revo worked for me! PHEW!

**Comment by DevDorrejo** [comment]
for those that revo/bcu or any uninstaller or other solution didn't work, with the tailscale team, we found this in my machine:
```
!    sto:           Unable to determine presence of driver package 'oem63.inf'. Error = 0x00000020
     inf:           {Query Configurability: c:\windows\inf\oem63.inf} 23:56:37.028
```

We eliminate this driver, and tailscale started to work normally, the command `pnputil.exe /d oem63.inf`

**Comment by chinhodado** [comment]
I managed to fix this by

- Uninstall Tailscale
- Delete `C:\Program Files\Tailscale`
- Delete `C:\ProgramData\Tailscale`
- Delete `C:\Users\xyz\AppData\Local\Tailscale`
- Re-install Tailscale

**Comment by onaka-yurusugi** [comment]
> I managed to fix this by
> 
> Uninstall Tailscale
> Delete C:\Program Files\Tailscale
> Delete C:\ProgramData\Tailscale
> Delete C:\Users\xyz\AppData\Local\Tailscale
> Re-install Tailscale

Same steps worked for me. Thanks!

**Comment by MovFish** [comment]
Is there any way to disable TPM authentication in win11? My device can't use TPM at all.

Now I can only downgrade Tailscale

**Comment by Cp0204** [comment]
> I managed to fix this by
> 
> * Uninstall Tailscale
> * Delete `C:\Program Files\Tailscale`
> * Delete `C:\ProgramData\Tailscale`
> * Delete `C:\Users\xyz\AppData\Local\Tailscale`
> * Re-install Tailscale

Thx, it works

**Comment by SlackingVeteran** [comment]
> I managed to fix this by
> 
>     Uninstall Tailscale
>     Delete C:\Program Files\Tailscale
>     Delete C:\ProgramData\Tailscale
>     Delete C:\Users\xyz\AppData\Local\Tailscale
>     Re-install Tailscale

You can simply stop Tailscale service, delete `C:\ProgramData\Tailscale` and restart Tailscale service. Dont have to uninstall and re-install the app

**Comment by Marsillio** [comment]
I had the same issue when my computer clock was incorrectly set, once I corrected the date the problem solved and tailscale worked properly!

**Comment by croakerbcts** [comment]
As the original reporter and multiple others have found a resolution, closing this issue.
